From 332b21b82c40df5bbb84728a59e703cfff1c112b Mon Sep 17 00:00:00 2001
From: Julian Scheel <julian@jusst.de>
Date: Mon, 23 Apr 2012 18:02:59 +0200
Subject: [PATCH] mpegtspacketizer: Disable use of clock skew.

When clock skew computation is used for computing timestamps it fails to
handle pts/dts rollovers correctly. For now disable usage of the clock skew to
workaround this issue.
---
 gst/mpegtsdemux/mpegtspacketizer.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/gst/mpegtsdemux/mpegtspacketizer.c b/gst/mpegtsdemux/mpegtspacketizer.c
index 41ff6dd..8cc66d1 100644
--- a/gst/mpegtsdemux/mpegtspacketizer.c
+++ b/gst/mpegtsdemux/mpegtspacketizer.c
@@ -3208,6 +3208,7 @@ mpegts_packetizer_pts_to_ts (MpegTSPacketizer2 * packetizer, GstClockTime pts)
 {
   GstClockTime res = GST_CLOCK_TIME_NONE;
 
+#if 0
   /* Use clock skew if present */
   if (packetizer->calculate_skew
       && GST_CLOCK_TIME_IS_VALID (packetizer->base_time)) {
@@ -3217,6 +3218,7 @@ mpegts_packetizer_pts_to_ts (MpegTSPacketizer2 * packetizer, GstClockTime pts)
     res = pts - packetizer->base_pcrtime + packetizer->base_time +
         packetizer->skew;
   } else
+#endif
     /* If not, use pcr observations */
   if (packetizer->calculate_offset && packetizer->priv->first_pcr != -1) {
     /* Rollover */
-- 
1.7.10

